src/services/services.ts

```diff
@@ -4316,6 +4361,159 @@ namespace ts {
                     }
                 }
             }
+
+            function getStringLiteralCompletionEntriesFromModuleNames(node: StringLiteral) {
+                const literalValue = node.text;
+                let result: CompletionEntry[];
+
+                const isRelativePath = startsWith(literalValue, ".");
+                const scriptDir = getDirectoryPath(node.getSourceFile().path);
+                if (isRelativePath || isRootedDiskPath(literalValue)) {
+                    result = getCompletionEntriesForDirectoryFragment(literalValue, scriptDir, getSupportedExtensions(program.getCompilerOptions()), /*includeExtensions*/false);
+                }
+                else {
+                    // Check for node modules
+                    result = getCompletionEntriesForNonRelativeModules(literalValue, scriptDir);
+                }
+
+                return {
+                    isMemberCompletion: false,
+                    isNewIdentifierLocation: false,
+                    entries: result
+                };
+            }
+
+            function getCompletionEntriesForDirectoryFragment(fragment: string, scriptPath: string, extensions: string[], includeExtensions: boolean): CompletionEntry[] {
+                // Complete the path by looking for source files and directories
+                const result: CompletionEntry[] = [];
+
+                const toComplete = getBaseFileName(fragment);

```

Comment:
we need to handle the case of requesting completion on a file with an extension... e.g.

```ts
import d from ".\foo.js| // we should complete this to `./foo.js` and not just `./foo`
```